source/initial_temperature/anelasticity_temperature.cc

```diff
@@ -0,0 +1,421 @@
+      // return the absolute temperature in Kelvin
+      return temperature;
+    }
+
+    // set up function to calculate shear wave velocity from Yamauchi & Takei 2016 parameterisation
+    template <int dim>
+    double
+    AnelasticVs2T<dim>::
+    yamauchi_takei_Vs (double temperature,
+                       double depth,
+                       const double mu0,
+                       const double dmudT,
+                       const double dmudP,
+                       const double viscosity_prefactor,
+                       const double activation_energy,
+                       const double activation_volume,
+                       const double solidus_gradient,
+                       const bool use_original_model) const
+    {
+      // specify anelasticity parameters
+      const double critical_homologous_temperature = 0.94;
+      const double reduction_factor = 5;
+      const double background_amplitude = 0.664;
+      const double background_slope = 0.38;
+      const double peak_period = 6e-5;
+      const double melt_viscosity_factor = 0;
+      const double melt_peak_factor = 0;
+      const double reference_temperature = 1473;
+      const double reference_pressure = 1.5e9;
+      const double grain_size = 1e-3;
+      const double reference_grain_size = 1e-3;
+      const double grain_size_exponent = 3;
+      const double pressure_gradient = 3e-5;
+      const double gas_constant=8.3145;
+      // specify Grose & Afonso (2013) density parameters
+      const double a=1;
+      const double b=3;
+      const double bulk_modulus=130e9;
+      const double bulk_modulus_pressure_derivative=4.8;
+      const double gruneisen_parameter=6;
+      const double reference_density=3330;
+      // specify original density parameters
+      const double original_density=3291;
+      const double original_thermal_expansivity=3.59e-5;
+      const double original_bulk_modulus=115.2;
+      // initialize solidus
+      const double T_solidus = 1326.0 + 273 + (((depth-50000)/1e3)*solidus_gradient);
+      // initialize homologous_temperature
+      double homologous_temperature = temperature/T_solidus;
+      // initialize pressures
+      double pressure = depth/pressure_gradient;
+      // declare other parameters
+      double viscosity,viscosity_reduction_factor,peak_amplitude,peak_width,isothermal_volume_change;
+      double compressibility,pressure_dependent_density,integrated_thermal_expansivity,density,anharmonic_Vs;
+      double unrelaxed_compliance,loss_compliance,storage_compliance,attenuation,period,anelastic_Vs;
+      // begin calculation of Vs
+      if (homologous_temperature<critical_homologous_temperature)
+        {
+          viscosity_reduction_factor=1;
+        }
+      else if ((homologous_temperature>=critical_homologous_temperature) && (homologous_temperature<1))
+        {
+          viscosity_reduction_factor=std::exp((-1*((homologous_temperature-critical_homologous_temperature)/(homologous_temperature-
```

Comment:
Here and in all other place, clearly separate integers from doubles. Not super important (as C++ automatically promotes integers), but important if you work in more than one language (makes your life easier), and clarifies the types of the variables in the equation (no need to worry if this might be an integer division).
```suggestion
          viscosity_reduction_factor=std::exp((-1.0*((homologous_temperature-critical_homologous_temperature)/(homologous_temperature-
```